### PR TITLE
Kim shellsort fp

### PR DESCRIPTION
**Pull request by KimSvenSand**

### CODE REVIEW

File: kim-shellsort-fp/src/js/general-functions.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+'use strict'
+
+function compose(f, g) {
```
Comment (Kazads) [inline kim-shellsort-fp/src/js/general-functions.js]: Unnecessary function

File: kim-shellsort-fp/es6-test/shellsort-functions-test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { shellsort, createGapSequence, insertionSort, sortSection, switchElements } from '../dist/js/shellsort-functions'
+import { expect } from 'chai'
+'use strict'
+
```
Comment (Kazads) [inline kim-shellsort-fp/es6-test/shellsort-functions-test.js:4]: Missing test on empty array

**Review by Kazads — COMMENTED** [review]
Some things missed, otherwise good

File: kim-shellsort-fp/src/js/shellsort-functions.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import { compose } from './general-functions'
+'use strict'
+
+function shellsort(array, gapIndex) {
+  var gapSequence = createGapSequence(array.length);
+  var copyArray = array.slice();
```
Comment (emilfolino) [inline kim-shellsort-fp/src/js/shellsort-functions.js:5]: Bra att ni kopierar.

**Review by emilfolino — COMMENTED** [review]
Bra följer alla regler.

**Comment by KimSvenSand** [comment]
Done
